lua/starfall/libs_sv/wire.lua

```diff
@@ -318,17 +318,29 @@ function wire_library.serverUUID()
 	return WireLib.GetServerUUID()
 end
 
+local ValidWireMat = { 	["cable/rope"] = true, ["cable/cable2"] = true, ["cable/xbeam"] = true, ["cable/redlaser"] = true, ["cable/blue_elec"] = true, ["cable/physbeam"] = true, ["cable/hydra"] = true, ["arrowire/arrowire"] = true, ["arrowire/arrowire2"] = true }
 --- Wires two entities together
 -- @param entI Entity with input
 -- @param entO Entity with output
 -- @param inputname Input to be wired
 -- @param outputname Output to be wired
-function wire_library.create (entI, entO, inputname, outputname)
+-- @param width Width of the wire(optional)
+-- @param color Color of the wire(optional)
+-- @param material Material of the wire(optional), Valid materials are cable/rope, cable/cable2, cable/xbeam, cable/redlaser, cable/blue_elec, cable/physbeam, cable/hydra, arrowire/arrowire, arrowire/arrowire2
+function wire_library.create (entI, entO, inputname, outputname, width, color, material)
 	checktype(entI, SF.Types["Entity"])
 	checktype(entO, SF.Types["Entity"])
 	checkluatype(inputname, TYPE_STRING)
 	checkluatype(outputname, TYPE_STRING)
-
+	
+	local width = width or 0
+	checkluatype(width, TYPE_NUMBER,0,0)
+	width = math.Clamp(width,0,5)
+	
+	checktype(color, SF.Types['Color'],0,SF.Color.Wrap(Color(255,0,0)))
```

Comment:
It returns default val but u never assign it. Also keep it consistent with other libs and just split it to:
```
color = color or SF.Color.Wrap(Color(255, 255, 255))
checktype(color, SF.Types['Color']) ```

Reply:
That's less efficient.